app/app.go

```diff
@@ -115,14 +117,18 @@ func MustNew(ctx context.Context, adminPort string) *App {
 }
 
 // Shutdown calls all shutdown methods, in order they were added.
-func (a *App) Shutdown(ctx context.Context) error {
+func (a *App) Shutdown(ctx context.Context) (err error) {
+	log.Trace().Msg("[app] Starting graceful shutdown")
+	defer log.Trace().Err(err).Msg("[app] Graceful shutdown finished")
```

Comment:
Nit: I'm not sure, but I think that `err` is evaluated now, not when executing the defer function.
https://play.golang.org/p/Q0QbyeGcTmv

Reply:
You are totally right. I've fixed this issue in a new commit and added a couple of more messages for make it easier debug the app livecycle.